builder/api/src/main/java/io/helidon/builder/api/Prototype.java

```diff
@@ -0,0 +1,414 @@
+                         Class<T> providerType,
+                         Class<S> configType,
+                         boolean allFromServiceLoader) {
+            return ProvidedUtil.discoverServices(config, serviceLoader, providerType, configType, allFromServiceLoader);
+        }
+    }
+
+    /**
+     * A prototype {@link Prototype.Blueprint} may extend this interface
+     * to explicitly reference the associated runtime type.
+     * <p>
+     * For example a {@code RetryPrototypeBlueprint} that extends a {@link Prototype.Factory}
+     * of {@code Retry}, will add methods to the prototype interface and builder to build an instance of
+     * {@code Retry} directly.
+     * A factory method must exist either on the runtime type (such as {@code Retry}) with signature
+     * {@code static Retry create(RetryPrototype)}, or on the prototype blueprint.
+     *
+     * @param <T> type of the runtime object (such as {@code Retry} in the description above)
+     */
+    interface Factory<T> {
+        /**
+         * Create a new instance of the runtime type from this config object.
+         *
+         * @return new configured runtime instance
+         */
+        T build();
+        /*
+        This method intentionally clashes with io.helidon.common.Builder.build(), as we need to have it
+        available both on the builder and on the prototype instance.
+         */
+    }
+
+    /**
+     * A package local type (by design) that defines getter methods and possible static factory methods
+     * that form prototype information that is generated through annotation processing.
+     * <p>
+     * The following names are prohibited from being builder properties:
+     * <ul>
+     *     <li>{@code build} - builder terminating method that builds runtime type from builder,
+     *                          and creates runtime type from prototype</li>
+     *     <li>{@code buildPrototype} - builder terminating method that builds prototype</li>
+     *     <li>{@code metaInf} - reserved for static method to obtain meta-information about this prototype</li>
+     *     <li>{@code config} - reserved for accessor for Helidon configuration object, if this type
+     *                          is {@code io.helidon.config.metadata.Configured} </li>
+     *     <li>{@code from} - reserved for a builder method to update it from existing prototype</li>
+     *     <li>{@code get} - same as {@code build}, inherited from {@link java.util.function.Supplier}</li>
+     * </ul>
+     */
+    @Target(ElementType.TYPE)
+    // note: class retention needed for cases when derived builders are inherited across modules
+    @Retention(RetentionPolicy.CLASS)
+    @interface Blueprint {
+        /**
+         * The generated interface is public by default. We can switch it to package local
+         * by setting this property to {@code false}-
+         * @return whether the generated interface should be public
+         */
+        boolean isPublic() default true;
```

Comment:
why the `is` prefix here - i thought that was against coding guidelines.

Reply:
try doing `public` as a method name ;)